x-pack/filebeat/input/streaming/config.go

```diff
@@ -73,6 +75,20 @@ type retry struct {
 	InfiniteRetries bool          `config:"infinite_retries"`
 }
 
+// KeepAlive is the configuration for keep-alive settings.
+type KeepAlive struct {
```

Comment:
Is there a reason this needs to be exported? If it's just to avoid name collision, rename this to `keepAliveConfig`.